mesonbuild/compilers/rust.py

```diff
@@ -108,7 +101,15 @@ def sanity_check(self, work_dir: str, environment: 'Environment') -> None:
         pe = subprocess.Popen(cmdlist, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
         pe.wait()
         if pe.returncode != 0:
-            raise EnvironmentException('Executables created by Rust compiler %s are not runnable.' % self.name_string())
+            raise EnvironmentException(f'Executables created by Rust compiler {self.name_string()} are not runnable.')
+        # Get libraries needed to link with a Rust staticlib
+        cmdlist = self.exelist + ['--crate-type', 'staticlib', '--print', 'native-static-libs', source_name]
+        p, stdo, stde = Popen_safe(cmdlist, cwd=work_dir)
+        if p.returncode == 0:
+            match = re.search('native-static-libs: (.*)$', stde, re.MULTILINE)
+            if match:
+                exclude = {'-lc', '-lgcc_s'}
```

Comment:
This could probably get a comment to explain why

Reply:
It probably does not hurt to have them, but I wasn't sure if `'-lgcc_s'` could cause issues if using that staticlib with clang for example? We usually don't have those flags in pkgconfig files:
```
$ pkg-config --static --libs glib-2.0
-lglib-2.0 -lm -pthread -lpcre2-8
```

Those flags gets automatically added when generating a pc file for a staticlib btw, which is good.

Reply:
As a matter of principle it should be the linker driver's job to figure out how to internally link to both of these support libraries when needed. It seems like a rust bug to emit them at all.

Reply:
Hmmm, maybe I should remove the exclude list and just call it a rustc bug if it causes any issue?

Reply:
Seems cheap enough to keep or what do you think?

Reply:
If, per the linked rustc ticket, it's indeed a bug to emit them and results in failed builds under certain conditions, then I think excluding them here counts as "working around a rustc bug so that meson can actually successfully build your code"?

Reply:
Added a FIXME with the link to Rust bug. I think we can merge it like that.

Reply:
@sdroege done